src/utils/viewer/viewer.cpp

```diff
@@ -352,14 +373,30 @@ void Viewer::drawEvent() {
     physicsManager_->debugDraw(projM * camM);
   }
 
+  if (displayPerformanceInfo) {
+    textRenderer_->draw(0xffffff_rgbf);
+  }
+
   swapBuffers();
-  timeline_.nextFrame();
   redraw();
+
+  totalFrames_++;
+  timeDuration_ += timeline_.previousFrameDuration();
```

Comment:
Note: now I used the `previousFrameDuration` not the `previousFrameTime`.
I simply compute the average FPS for every `fpsUpdateFrequency`. 

You would like to have a moving window (window size = fpsUpdateFrequency)? Then I have to implement a small ring buffer (buffer size = fpsUpdateFrequency), and more importantly, that means I need to update the text every frame (otherwise, if I update it in every `fpsUpdateFrequency`, this is what I am doing here.)

Reply:
Yeah, that's why I suggested to have all this implemented on the side of Magnum -- because it's not feasible to have ring buffer, Kahan Sum, moving average and all that implemented & tested here just for the needs of a FPS counter of a viewer utility.

Let's keep it as you have it now, we can always revisit later.

Reply:
Make sense.
By the way, in my previous projects, we simply use double to increase the precision, and never think Kahan summation is needed. Maybe it is because my old projects were computation heavy and thus the FPS was too low. :-) Have you experienced any case that it is a must-have? Just curious. I hope to understand and learn.

Reply:
Probably not so important in this case where we have a dozen values to average each time, but would definitely make a difference when calculating an average FPS over the whole application run time -- with floats, it'd start losing precision in the microsecond range already after 15 minutes. With other measured quantities such as sample fill rate or primitive emission it could be even sooner. To give you a perspective, [in this test](https://github.com/mosra/magnum/blob/debddf658571cd2a653c6ffe82bc39f245a51ba7/src/Magnum/Math/Algorithms/Test/KahanSumTest.cpp#L87-L100) I sum `1.0f` 100 million times and while Kahan Sum gives me 1.0e8, the naive approach *stops* at 1.66667e7 and doesn't go higher because the newly added values get rounded off during addition.

Doubles would be sufficient for precise calculations when the app is running days, but not much more I think.